test/e2e/functional/tests/e2e.go

```diff
@@ -208,8 +208,8 @@ var _ = Describe("metallb", func() {
 			})
 
 			AfterEach(func() {
-				metallbutils.Delete(incorrect_metallb)
 				metallbutils.Delete(correct_metallb)
+				metallbutils.Delete(incorrect_metallb)
```

Comment:
I think the right way to do is to revise Delete and have it "just" delete, and moving the check logic in another function, OR having a `Delete` and a `DeleteAndCheck`

Reply:
done